cai_causal_graph/causal_graph.py

```diff
@@ -1726,6 +1726,49 @@ def get_common_descendants(self, node_1: NodeLike, node_2: NodeLike) -> Set[str]
 
         return self.get_descendants(node_1).intersection(self.get_descendants(node_2))
 
+    def get_nodes_between(self, node_1: NodeLike, node_2: NodeLike) -> Set[Node]:
+        """
+        Get the set of nodes that are on causal paths between two nodes.
```

Comment:
Code only seems to work for directed paths, right? I would make this clear, e.g. by stating "directed causal paths" but also be explicit in a sentence below that this will not pick up on nodes connected via, e.g., undirected edges.